keboola/component_configuration_resource.go

```diff
@@ -0,0 +1,426 @@
+			"Error creating configuration",
+			"Could not create configuration: "+err.Error(),
+		)
+		return
+	}
+	plan.ConfigID = types.StringValue(resConfig.ID.String())
+	plan.BranchID = types.Int64Value(int64(resConfig.BranchID))
+	plan.IsDeleted = types.BoolValue(resConfig.IsDeleted)
+	plan.ID = types.StringValue(getConfigModelId(&plan))
+	plan.Version = types.Int64Value(int64(resConfig.Version))
+	plan.Created = types.StringValue(resConfig.Created.String())
+	plan.ChangeDescription = types.StringValue(resConfig.ChangeDescription)
+	plan.Description = types.StringValue(resConfig.Description)
+	plan.IsDisabled = types.BoolValue(resConfig.IsDisabled)
+
+	// Set state to fully populated data
+	diags = resp.State.Set(ctx, plan)
+
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read resource information
+func (r *configResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	// Get current state
+	var state configModel
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get refreshed configuration
+	key := ConfigKey{
+		ID:          ConfigID(state.ConfigID.ValueString()),
+		BranchID:    BranchID(state.BranchID.ValueInt64()),
+		ComponentID: ComponentID(state.ComponentID.ValueString()),
+	}
+	config, err := GetConfigRequest(key).Send(ctx, r.sapiClient)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error Reading Configuration",
+			"Could not read Configuration"+getConfigModelId(&state)+": "+err.Error(),
+		)
+		return
+	}
+
+	// Overwrite config with refreshed state
+	state.Name = types.StringValue(config.Name)
+	state.Description = types.StringValue(config.Description)
+	state.ChangeDescription = types.StringValue(config.ChangeDescription)
+	state.IsDeleted = types.BoolValue(config.IsDeleted)
+	state.Created = types.StringValue(config.Created.String())
+	state.Version = types.Int64Value(int64(config.Version))
+	state.IsDisabled = types.BoolValue(config.IsDisabled)
+	state.ID = types.StringValue(getConfigModelId(&state))
+
+	currentContentMap := orderedmap.New()
+	err = currentContentMap.UnmarshalJSON([]byte(state.Content.ValueString()))
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error reading configuration",
+			"Could not parse state configuration: "+err.Error(),
+		)
+		return
+	}
+
+	newContentMap := config.Content
+	newContentBytes, err := newContentMap.MarshalJSON()
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error Reading Configuration",
+			"Could not parse read configuration "+getConfigModelId(&state)+": "+err.Error(),
+		)
+		return
+	}
+	newContentStr := string(newContentBytes)
+
+	if !reflect.DeepEqual(currentContentMap, newContentMap) {
+		resp.Diagnostics.AddWarning(
+			"Read configuration changed",
+			"Updating local state with to match the read configuration",
+		)
+		state.Content = types.StringValue(newContentStr)
+	}
```

Comment:
configy su definovane ako stringy ale kbc api vracia config content bez whitespaces a preto si to terraform vylozil ako 2 rozne stavy a vracal chybu. Tak som to musel porovnavat ako orderedmaps cez DeepEqual